### DISCUSSION TITLE
Push towards `primitives v1.0`

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#3635] [General]
`primitives 0.101.0` is out - the first _real_ release. But no rest for the wicked. Now we have to push towards `primitives 1.0`. That means we need to 1.0 all the crates depended on by `primitives`. This is all tracked here: https://github.com/rust-bitcoin/rust-bitcoin/issues/3133

We have an issue (#1649) that @Kixunil created ages ago for when we were getting close - that is now. As such I've duplicated the issue once for each crate, shall we start chipping away at them and as each item is done for a crate we check it off?

I've added a link to the duplicated issue on the respective tracking issue for each crate, or filter on 'crate conform'.

Thanks

**Comment by apoelstra** [comment]
This plan sounds great. Thanks for doing this!

**Comment by tcharding** [comment]
After creating all the individual 'Make crate conform to API guidelines' I realized that for some things its quicker/eaiser to check them all at once (eg `C-PERMISSIVE`). I am marking these on the, now titled, 'Repository Wide: Make crate conform to API guidelines'
 
https://github.com/rust-bitcoin/rust-bitcoin/issues/1649
